### PR TITLE
Fix syntax on ChefCorrectness/IncorrectLibraryInjection.

### LINKED ISSUE #597: ChefCorrectness/IncorrectLibraryInjection incorrect syntax
# Version:

# Environment:

# Scenario:

ChefCorrectness/IncorrectLibraryInjection uses `Chef::DSL::Resource` as the replacement for `Chef::Resource`.  This should be `Chef::DSL::Resources`.

# Steps to Reproduce:
From chef-shell, 

# Expected Result:

```
chef (15.8.23)> ::Chef::DSL::Resources.send(:include, Filebeat::Helpers)
Traceback (most recent call last):
        8: from /opt/chef-workstation/bin/chef-shell:362:in `<main>'
        7: from /opt/chef-workstation/bin/chef-shell:362:in `load'
        6: from /opt/chef-workstation/embedded/lib/ruby/gems/2.6.0/gems/chef-bin-15.8.23/bin/chef-shell:31:in `<top (required)>'
        5: from /opt/chef-workstation/embedded/lib/ruby/gems/2.6.0/gems/chef-15.8.23/lib/chef/shell.rb:81:in `start'
        4: from /opt/chef-workstation/embedded/lib/ruby/gems/2.6.0/gems/chef-15.8.23/lib/chef/shell.rb:81:in `catch'
        3: from /opt/chef-workstation/embedded/lib/ruby/gems/2.6.0/gems/chef-15.8.23/lib/chef/shell.rb:82:in `block in start'
        2: from (irb):3
        1: from (irb):3:in `rescue in irb_binding'
NameError (uninitialized constant Filebeat)
```
Notice the `::Chef::DSL::Resources` constant is initialized.  This error is expected, since I don't have a Filebeat constant.

# Actual Result:

```
chef (15.8.23)> ::Chef::DSL::Resource.send(:include, Filebeat::Helpers)
Traceback (most recent call last):
        8: from /opt/chef-workstation/bin/chef-shell:362:in `<main>'
        7: from /opt/chef-workstation/bin/chef-shell:362:in `load'
        6: from /opt/chef-workstation/embedded/lib/ruby/gems/2.6.0/gems/chef-bin-15.8.23/bin/chef-shell:31:in `<top (required)>'
        5: from /opt/chef-workstation/embedded/lib/ruby/gems/2.6.0/gems/chef-15.8.23/lib/chef/shell.rb:81:in `start'
        4: from /opt/chef-workstation/embedded/lib/ruby/gems/2.6.0/gems/chef-15.8.23/lib/chef/shell.rb:81:in `catch'
        3: from /opt/chef-workstation/embedded/lib/ruby/gems/2.6.0/gems/chef-15.8.23/lib/chef/shell.rb:82:in `block in start'
        2: from (irb):2
        1: from (irb):2:in `rescue in irb_binding'
NameError (uninitialized constant Chef::DSL::Resource)
```
Notice the uninitialized constant for `Chef::DSL::Resource`.

### PR DESCRIPTION
**Pull request by ncerny**
Signed-off-by: Nathan Cerny <ncerny@chef.io>

## Description
Fixes syntax issue in ChefCorrectness/IncorrectLibraryInjection cop.

## Related Issue
Fixes #597 

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by ncerny** [comment]
The Buildkite job does not seem to run the test for `RuboCop::cop::chef::ChefCorrectness::IncorrectLibraryInjection`

File: lib/rubocop/cop/chef/correctness/incorrect_library_injection.rb [hunk 1]
```diff
@@ -29,10 +29,10 @@ module ChefCorrectness
         #
         #   # good
         #   ::Chef::DSL::Recipe.send(:include, Filebeat::Helpers) # covers previous Recipe & Provider classes
-        #   ::Chef::DSL::Resource.send(:include, Filebeat::Helpers)
+        #   ::Chef::DSL::Resources.send(:include, Filebeat::Helpers)
         #
         class IncorrectLibraryInjection < Cop
-          MSG = 'Libraries should be injected into the Chef::DSL::Recipe or Chef::DSL::Resource classes and not Recipe/Resource/Provider classes directly.'.freeze
+          MSG = 'Libraries should be injected into the Chef::DSL::Recipe or Chef::DSL::Resources classes and not Recipe/Resource/Provider classes directly.'.freeze
```
Comment (lamont-granquist) [inline lib/rubocop/cop/chef/correctness/incorrect_library_injection.rb:35]: That should be Chef::DSL::Universal, the Chef::DSL::Resources class is marked `@api private` and injecting a library into that class probably just won't work (it lacks the LazyModuleInclude helper that are in the Universal + Recipe DSLs.

File: docs/cops_chefcorrectness.md [hunk 2]
```diff
@@ -169,7 +169,7 @@ Enabled by default | Supports autocorrection
 --- | ---
 Enabled | Yes
 
-Libraries should be injected into the Chef::DSL::Recipe or Chef::DSL::Resource classes and not Recipe/Resource/Provider classes directly.
```
Comment (tas50) [inline docs/cops_chefcorrectness.md:172]: @ncerny if you end up in Cookstyle land again these are actually auto generated from the source so you don't need to update them.

**Comment by tas50** [comment]
Updated based on a conversation in Slack

**Pull request merged**